blockchain/validate.go

```diff
@@ -821,6 +841,44 @@ func CheckBlockSanity(block *dcrutil.Block, timeSource MedianTimeSource, chainPa
 	return checkBlockSanity(block, timeSource, BFNone, chainParams)
 }
 
+// isDCP0005Violation returns whether or not the block is known to violate
+// DCP0005.  Due to a bug that has since been fixed, some blocks were accepted
+// that violated DCP0005 by still being submitted on an old block version
+// (version 6 on mainnet) when the majority of the network had already upgraded
+// to the new version specified by DCP0005 (version 7 on mainnet).
+//
+// The blocks that violated DCP0005 due to specifying the old version were
+// otherwise valid and are whitelisted by this function so that they will be
+// accepted when fully syncing the chain without checkpoints.
+func isDCP0005Violation(network wire.CurrencyNet, header *wire.BlockHeader, blockHash *chainhash.Hash) bool {
+	// 430191 is the height of the last block on mainnet that violated DCP0005, so
+	// return false immediately if the height is greater than that.
+	if header.Height > 430191 {
```

Comment:
I can update it to that.  I actually had something similar at first and then removed it to avoid the extra network check, but in practice, it shouldn't really have any performance impact, and this reads better.